### PR TITLE
sqld: 0.17.2 -> 0.24.18

### PR DESCRIPTION
**Pull request by rawkode**
## Description of changes

Following up after #290473 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by rawkode** [comment]
I don't think this error is related to my change. 

cc @alyssais who referenced this is kind of important due to Rust updates in staging.

**Comment by alyssais** [comment]
Please rebase on latest master to fix CI.

File: pkgs/servers/sql/sqld/default.nix [hunk 1]
```diff
@@ -1,35 +1,45 @@
...
   };
 
+  cargoBuildFlags = [
+    "--bin"
+    "sqld"
+  ];
+
   cargoLock = {
-    lockFile = ./Cargo.lock;
+    lockFile = "${src}/Cargo.lock";
```
Comment (alyssais) [inline pkgs/servers/sql/sqld/default.nix:32]: This is import-from-derivation, which is not allowed in Nixpkgs.

File: pkgs/servers/sql/sqld/default.nix [hunk 2]
```diff
@@ -1,35 +1,45 @@
...
-      "octopod-0.1.0" = "sha256-V16fOlIp9BCpyzgh1Aei3Mra/y15v8dQFA8tHdOwZm4=";
+      "console-api-0.5.0" = "sha256-MfaxtzOqyblk6aTMqJGRP+123aK0Kq7ODNp/3lehkpQ=";
+      "hyper-rustls-0.24.1" = "sha256-dYN42bnbY+4+etmimrnoyzmrKvCZ05fYr1qLQFvzRTk=";
+      "rheaper-0.2.0" = "sha256-u5z6J1nmUbIQjDDqqdkT0axNBOvwbFBYghYW/r1rDHc=";
+      "s3s-0.10.1-dev" = "sha256-y4DZnRsQzRNsCIp6vrppZkfXSP50LCHWYrKRoIHYPik=";
     };
   };
 
   nativeBuildInputs = [
+    cmake
```
Comment (alyssais) [inline pkgs/servers/sql/sqld/default.nix:42]: What's this for?  This usually indicates a vendored C/C++ library is being built, which we usually don't want (it's better to get the library from Nixpkgs).

**Review by alyssais — CHANGES_REQUESTED** [review]

File: pkgs/servers/sql/sqld/default.nix (hunk 2 above)
Comment (rawkode) [reply pkgs/servers/sql/sqld/default.nix:42]: It's the only way I could get it to build. I'm probably out of my depth here then.

Comment (alyssais) [reply pkgs/servers/sql/sqld/default.nix:42]: Sadly it looks like libsql-ffi doesn't support building any other way, so I guess we can do this.

File: pkgs/servers/sql/sqld/default.nix (hunk 1 above)
Comment (rawkode) [reply pkgs/servers/sql/sqld/default.nix:32]: What's the best way to do this then?

Comment (rawkode) [reply pkgs/servers/sql/sqld/default.nix:32]: I copied it from another package 😅

Comment (alyssais) [reply pkgs/servers/sql/sqld/default.nix:32]: Which other package?  I don't see any other occurrence of `lockFile = "${src}` in Nixpkgs, and it shouldn't get past CI.

The right way to do it is the way it was done before, by including Cargo.lock in Nixpkgs.

Comment (rawkode) [reply pkgs/servers/sql/sqld/default.nix:32]: Oh, it was a search for `lang:nix lockfile`, so it wasn't from nixpkgs; my mistake.

Rebasing and fixing now.

**Comment by rawkode** [comment]
@alyssais Hopefully this resolves the issues

**Comment by rawkode** [comment]
Oh, and thank you very much for your help!

**Review by alyssais — APPROVED** [review]

**Comment by alyssais** [comment]
I'm afraid you'll have to rebase again to pick up https://github.com/NixOS/nixpkgs/pull/337041.

**Comment by rawkode** [comment]
> I'm afraid you'll have to rebase again to pick up https://github.com/NixOS/nixpkgs/pull/337041.

Done

**Comment by alyssais** [comment]
```
error: builder for '/nix/store/00raa9m95cjc6iafn442y2j1y4z6lrni-source.drv' failed with exit code 1;
       last 8 log lines:
       >
       > trying https://github.com/tursodatabase/libsql/archive/libsql-server-v0.24.18g.tar.gz
       >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
       >                                  Dload  Upload   Total   Spent    Left  Speed
       >   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       >   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       > curl: (22) The requested URL returned error: 404
       > error: cannot download source from any mirror
```

Hmm, that's not good

File: pkgs/servers/sql/sqld/default.nix [hunk 3]
```diff
@@ -1,35 +1,47 @@
...
+  zstd,
+  stdenv,
+  darwin,
+  cmake,
 }:
 
 rustPlatform.buildRustPackage rec {
   pname = "sqld";
-  version = "0.17.2";
+  version = "0.24.18g";
```
Comment (alyssais) [inline pkgs/servers/sql/sqld/default.nix:17]: I guess this is a typo?

**Comment by rawkode** [comment]
Bah. Typo g

**Review by alyssais — CHANGES_REQUESTED** [review]

**Comment by rawkode** [comment]
Fixed

**Review by alyssais — APPROVED** [review]

**Review by alyssais — CHANGES_REQUESTED** [review]
```
trying https://github.com/tursodatabase/libsql/archive/libsql-server-v0.24.18.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 27.5M    0 27.5M    0     0  6531k      0 --:--:--  0:00:04 --:--:-- 7432k
unpacking source archive /build/libsql-server-v0.24.18.tar.gz
error: hash mismatch in fixed-output derivation '/nix/store/yxgrrsg0l2lwgyr965s688zxkyi6bg50-source.drv':
         specified: sha256-C9ze9o23n7v4NOotuKnOksK90FbhU0P+Rfnt64NZTQ8=
            got:    sha256-/0Sk55GBjUk/FeIoq/hGVaNGB0EM8CxygAXZ+ufvWKg=
```

**Comment by rawkode** [comment]
I must have messed up my rebase for that to be incorrect.

Resolved again.

**Review by alyssais — APPROVED** [review]

**Pull request merged**